stacks-signer/src/runloop.rs

```diff
@@ -13,1063 +13,234 @@
-                    "signer_id" => self.signer_id,
-                );
-                self.commands.push_back(RunLoopCommand::Sign {
-                    block: block_info.block.clone(),
-                    is_taproot: false,
-                    merkle_root: None,
-                });
-            } else {
-                debug!(
-                    "Signer ignoring block.";
-                    "block_hash" => block_info.block.header.block_hash(),
-                    "valid" => block_info.valid,
-                    "signed_over" => block_info.signed_over,
-                    "coordinator_id" => coordinator_id,
-                    "signer_id" => self.signer_id,
-                );
-            }
+        reward_cycle: u64,
+    ) -> Result<Option<RewardCycleConfig>, ClientError> {
+        let reward_set_calculated = self.stacks_client.reward_set_calculated(reward_cycle)?;
+        if !reward_set_calculated {
+            // Must weight for the reward set calculation to complete
+            // Accounts for Pre nakamoto by simply using the second block of a prepare phase as the criteria
+            return Err(ClientError::RewardSetNotYetCalculated(reward_cycle));
         }
-    }
-
-    /// Handle signer messages submitted to signers stackerdb
-    fn handle_signer_messages(
-        &mut self,
-        res: Sender<Vec<OperationResult>>,
-        messages: Vec<SignerMessage>,
-    ) {
-        let (coordinator_id, coordinator_public_key) =
-            calculate_coordinator(&self.signing_round.public_keys, &self.stacks_client);
-        debug!(
-            "Signer #{}: coordinator is signer #{} public key {}",
-            self.signer_id, coordinator_id, &coordinator_public_key
-        );
-        let packets: Vec<Packet> = messages
-            .into_iter()
-            .filter_map(|msg| match msg {
-                SignerMessage::BlockResponse(_) | SignerMessage::Transactions(_) => None,
-                SignerMessage::Packet(packet) => {
-                    self.verify_packet(packet, &coordinator_public_key)
-                }
-            })
-            .collect();
-        self.handle_packets(res, &packets);
-    }
-
-    /// Handle proposed blocks submitted by the miners to stackerdb
-    fn handle_proposed_blocks(&mut self, blocks: Vec<NakamotoBlock>) {
-        for block in blocks {
-            // Store the block in our cache
-            self.blocks.insert(
-                block.header.signer_signature_hash(),
-                BlockInfo::new(block.clone()),
+        let current_addr = self.stacks_client.get_signer_address();
+
+        let signer_set =
+            u32::try_from(reward_cycle % 2).expect("FATAL: reward_cycle % 2 exceeds u32::MAX");
+        let signer_stackerdb_contract_id =
+            boot_code_id(SIGNERS_NAME, self.config.network.is_mainnet());
+        // Get the signer writers from the stacker-db to find the signer slot id
+        let Some(signer_slot_id) = self
```

Comment:
@obycode this is the actual signer index. so you will need to do something similar to what I do in get_coordinator_sender in my signer test and this logic to get the signer index